app/assets/javascripts/students/components/CourseList.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { Link } from 'react-router'
+
+import { APIRoutes } from '../../shared/routes'
+import request from '../../shared/requests/request'
+import { ReactRoutes } from '../../shared/routes'
+
+import CourseCard from './CourseCard'
+
+class CourseList extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      courses: [],
+    }
+
+    this.getCourses()
```

Comment:
Theoretically you shouldn't call a function that will call `setState` in the constructor, you should wait on `componentDidMount` (but this should be ok).